librz/core/cmd_debug.c

```diff
@@ -2949,603 +2880,47 @@ RZ_IPI void rz_core_static_debug_stop(void *u) {
-				free(spstr);
-			}
-			rz_list_free(list);
-			break;
-		case '?':
-		default:
-			rz_core_cmd_help(core, help_msg_dbt);
-			break;
-		}
-		break;
-	case 'b': // "dbb"
-		if (input[2]) {
-			core->dbg->bp->delta = (st64)rz_num_math(core->num, input + 2);
-		} else {
-			rz_cons_printf("%" PFMT64d "\n", core->dbg->bp->delta);
-		}
-		break;
-	case 'm': // "dbm"
-		if (input[2] && input[3]) {
-			char *string = strdup(input + 3);
-			char *module = NULL;
-			st64 delta = 0;
-
-			module = strtok(string, " ");
-			delta = (ut64)rz_num_math(core->num, strtok(NULL, ""));
-			bpi = rz_debug_bp_add(core->dbg, 0, hwbp, false, 0, module, delta);
-			if (!bpi) {
-				eprintf("Cannot set breakpoint.\n");
-			}
-			free(string);
-		}
-		break;
-	case 'j': rz_bp_list(core->dbg->bp, 'j'); break;
-	case '*': rz_bp_list(core->dbg->bp, 1); break;
-	case '\0': rz_bp_list(core->dbg->bp, 0); break;
-	case '-': // "db-"
-		if (input[2] == '*') {
-			rz_bp_del_all(core->dbg->bp);
-		} else {
-#define DB_ARG(x) rz_str_word_get0(str, x)
-			char *str = strdup(rz_str_trim_head_ro(input + 2));
-			int i = 0;
-			int sl = rz_str_word_set0(str);
-			for (; i < sl; i++) {
-				const ut64 addr = rz_num_math(core->num, DB_ARG(i));
-				rz_bp_del(core->dbg->bp, addr);
-			}
-			free(str);
-		}
-		break;
-	case 'c': // "dbc"
-		if (input[2] == ' ') {
-			char *inp = strdup(input + 3);
-			if (inp) {
-				char *arg = strchr(inp, ' ');
-				if (arg) {
-					*arg++ = 0;
-					addr = rz_num_math(core->num, inp);
-					bpi = rz_bp_get_at(core->dbg->bp, addr);
-					if (bpi) {
-						free(bpi->data);
-						bpi->data = strdup(arg);
-					} else {
-						eprintf("No breakpoint defined at 0x%08" PFMT64x "\n", addr);
-					}
-				} else {
-					eprintf("- Missing argument\n");
-				}
-				free(inp);
-			} else {
-				eprintf("Cannot strdup. Your heap is a mess\n");
-			}
-		} else {
-			eprintf("Use: dbc [addr] [command]\n");
-		}
-		break;
-	case 'C': // "dbC"
-		if (input[2] == ' ') {
-			char *inp = strdup(input + 3);
-			if (inp) {
-				char *arg = strchr(inp, ' ');
-				if (arg) {
-					*arg++ = 0;
-					addr = rz_num_math(core->num, inp);
-					bpi = rz_bp_get_at(core->dbg->bp, addr);
-					if (bpi) {
-						free(bpi->cond);
-						bpi->cond = strdup(arg);
-					} else {
-						eprintf("No breakpoint defined at 0x%08" PFMT64x "\n", addr);
-					}
-				} else {
-					eprintf("1 Missing argument\n");
-				}
-				free(inp);
-			} else {
-				eprintf("Cannot strdup. Your heap is a mess\n");
-			}
-		} else {
-			eprintf("Use: dbC [addr] [command]\n");
-		}
-		break;
-	case 's': // "dbs"
-		addr = rz_num_math(core->num, input + 2);
-		rz_core_debug_breakpoint_toggle(core, addr);
-		break;
-	case 'n': // "dbn"
-		bpi = rz_bp_get_at(core->dbg->bp, core->offset);
-		if (input[2] == ' ') {
-			if (bpi) {
-				free(bpi->name);
-				bpi->name = strdup(input + 3);
+	addr = rz_num_math(core->num, arg_addr);
+	bpi = rz_debug_bp_add(core->dbg, addr, hwbp, watch, rw, NULL, 0);
+	if (bpi) {
```

Comment:
Here I think it's better to reverse the condition. Where possible we usually prefer control flow like that:
```c
if (!cond1) {
    return ERR1;
}
do_something();
if (!cond2) {
    return ERR2;
}
do_something2();
...
```
This way it's easier to track the proper error handling, and avoid deep nesting.